### PR TITLE
Enable building with cargo profiles

### PR DESCRIPTION
**Pull request by svenrademakers**
Hello, thank you for processing my recent PR's!

so my previous PR only supports `--profile` with `--no-build`. this PR enables to actually build the profile passed on to the CLI options.

e.g. you can do for instance 'cargo deb -p my_package --manifest-path applications/my_app/Cargo.toml --profile RelWithDebInfo --no-strip'

The other guy from the issue list will probably be happy to get it as well.

PS; you probably already know, but there is quite some technical debt in the build flags support.. i shouldnt really need to explicitly pass `--no-strip` if my cargo profile states that i want debug symbols

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -89,11 +89,11 @@ pub fn remove_deb_temp_directory(options: &Config) {
     let _ = fs::remove_dir(&deb_temp_dir);
 }
 
-/// Builds a release binary with `cargo build --release`
+/// Builds a binary with `cargo build`
 pub fn cargo_build(options: &Config, target: Option<&str>, other_flags: &[String], verbose: bool) -> CDResult<()> {
     let mut cmd = Command::new("cargo");
     cmd.current_dir(&options.manifest_dir);
-    cmd.arg("build").args(&["--release", "--all"]);
```
Comment (kornelski) [inline src/lib.rs:96]: This looks like a breaking change. We need `--all` to allow packaging multiple binaries/libraries from a workspace.

File: src/main.rs [hunk 2]
```diff
@@ -93,7 +99,8 @@ fn main() {
 
 #[allow(deprecated)]
 fn err_cause(err: &dyn std::error::Error, max: usize) {
-    if let Some(reason) = err.cause() { // we use cause(), not source()
+    if let Some(reason) = err.cause() {
+        // we use cause(), not source()
```
Comment (kornelski) [inline src/main.rs:103]: Please don't use rustfmt. The formatting in this crate is deliberate, and `rustfmt` makes it worse

File: src/lib.rs (hunk 1 above)
Comment (svenrademakers) [reply src/lib.rs:96]: sure! i have no opinion on this. wouldn't you want to pass the same manifest file as in the CLIOptions?

File: src/main.rs (hunk 2 above)
Comment (svenrademakers) [reply src/main.rs:103]: i see, will revert

File: src/lib.rs (hunk 1 above)
Comment (kornelski) [reply src/lib.rs:96]: I'm not sure what you mean?

Note that current dir is set to the dir that has the specified manifest.

**Comment by kornelski** [comment]
Thanks

**Pull request merged**